doc/guides/west/zephyr-cmds.rst

```diff
@@ -50,3 +50,74 @@ able to find the Zephyr repository with the following:
    find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 
 See :zephyr_file:`share/zephyr-package/cmake` for details.
+
+Software bill of materials: ``west spdx``
+*****************************************
+
+This command generates SPDX 2.2 tag-value documents, creating relationships
+from source files to the corresponding generated build files.
+``SPDX-License-Identifier`` comments in source files are scanned and filled
+into the SPDX documents.
+
+To use this command:
+
+#. Pre-populate a build directory :file:`BUILD_DIR` like this:
+
+   .. code-block:: bash
+
+      west spdx --init -d BUILD_DIR
+
+   This step ensures the build directory contains CMake metadata required for
+   SPDX document generation.
+
+#. Build your application using this pre-created build directory, like so:
+
+   .. code-block:: bash
+
+      west build -d BUILD_DIR [...]
+
+#. Generate SPDX documents using this build directory:
+
+   .. code-block:: bash
+
+      west spdx -d BUILD_DIR
+
+This generates the following SPDX bill-of-materials (BOM) documents in
+:file:`BUILD_DIR/spdx/`:
+
+- :file:`app.spdx`: BOM for the application source files used for the build
+- :file:`zephyr.spdx`: BOM for the specific Zephyr source code files used for
+   the build
+- :file:`build.spdx`: BOM for the built output files
+
+Each file in the bill-of-materials is scanned, so that its hashes (SHA256 and
+SHA1) can be recorded, along with any detected licenses if an
+``SPDX-License-Identifier`` comment appears in the file.
+
+SPDX Relationships are created to indicate dependencies between
+CMake build targets, build targets that are linked together, and
+source files that are compiled to generate the built library files.
+
+``west spdx`` accepts these additional options:
+
+- ``-n PREFIX``: a prefix for the Document Namespaces that will be included in
+  the generated SPDX documents. See `SPDX specification 2.2 section 2.5`_ for
+  details. If ``-n`` is omitted, a default namespace will be generated
+  according to the default format described in section 2.5 using a random UUID.
+
+- ``-s SPDX_DIR``: specifies an alternate directory where the SPDX documents
+  should be written instead of :file:`BUILD_DIR/spdx/`.
+
+- ``--analyze-includes``: in addition to recording the compiled source code
+  files (e.g. ``.c``, ``.S``) in the bills-of-materials, also attempt to
+  determine the specific header files that are included for each ``.c`` file.
+
+  This takes longer, as it performs a dry run using the C compiler for each
+  ``.c`` file using the same arguments that were passed to it for the actual
+  build.
+
+- ``--include-sdk``: with ``--analyze-includes``, also create a fourth SPDX
```

Comment:
@swinslow one thing I am not clear about with this option is if by "SDK" you really mean "toolchain". If so, we should probably also link to the toolchain related documentation so users are more certain. (If not, I think more clarification would be helpful.)

Reply:
Hi @mbolivar-nordic, I'm still learning a number of things about the Zephyr build process -- specifically, I've only run Zephyr builds via installing the Zephyr SDK toolchain as described in the [Getting Started docs](https://docs.zephyrproject.org/latest/getting_started/index.html#install-a-toolchain). So my answer might be incomplete :)

But I believe the way this currently works is that `--include-sdk` is specifically tied to installing the Zephyr SDK toolchain, not using an alternate toolchain. This is because it looks at `CMakeCache.txt` to find the `ZEPHYR_SDK_INSTALL_DIR` path variable, as part of building the SPDX document for included files from that directory. This happens at: https://github.com/zephyrproject-rtos/zephyr/blob/3d39f72a88b3100ac83e7c6c285ae567aca640d0/scripts/west_commands/zspdx/walker.py#L104-L110

This could presumably be improved in the future and made more flexible, for folks who aren't using the Zephyr SDK toolchain. For now they can still use `west spdx` anyway, and I think even the `--analyze-includes` flag should work for them; it just might not create SPDX entries for the files included from the toolchain in that case.

Reply:
To your point about clarifying -- the next line below (line 122) could perhaps be changed so that instead of ending with "the SDK", it instead says: "the Zephyr Software Development Kit (for builds where the SDK is being used)"

Reply:
Thanks for clarifying @swinslow! I think your suggested edit makes sense, but it's not urgent since it really does mean 'Zephyr SDK'.

If you want to start supporting other toolchains, you'll probably have to start looking at their specific configuration variables. E.g. for GNU Arm Embedded, you might need to test if `ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb` and then check `GNUARMEMB_TOOLCHAIN_PATH`. See https://docs.zephyrproject.org/latest/getting_started/toolchain_3rd_party_x_compilers.html#gnu-arm-embedded.

If it's not too much work, I suspect at least handling that toolchain is worthwhile, as I imagine the majority of Windows users targeting Cortex-M are using that configuration.